frontend/src/scenes/data-management/events/DefinitionHeader.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import { EventDefinition, PropertyDefinition } from '~/types'
+import {
+    AutocaptureIcon,
+    PageleaveIcon,
+    PageviewIcon,
+    PropertyIcon,
+    UnverifiedEventStack,
+    VerifiedEventStack,
+} from 'lib/components/icons'
+import { keyMapping, PropertyKeyInfo } from 'lib/components/PropertyKeyInfo'
+import { Tooltip } from 'lib/components/Tooltip'
+
+export enum DefinitionType {
+    Event = 'event',
+    Property = 'property',
+}
+
+export function getDefinitionIcon(
+    definition: EventDefinition | PropertyDefinition,
+    type: DefinitionType = DefinitionType.Event
+): JSX.Element {
+    if (type === DefinitionType.Property) {
+        return <PropertyIcon className="taxonomy-icon taxonomy-icon-muted" />
+    }
+    const _definition = definition as EventDefinition
```

Comment:
I think you can avoid this `as` cast with either overloading the function definition or using some kind of type guard function... 🤔